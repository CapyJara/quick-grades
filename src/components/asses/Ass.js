import React from 'react';
import PropTypes from 'prop-types';
import styles from './ass.css';
import { Link } from 'react-router-dom';

function Ass({ ass }) {
  return (
    <li className={styles.ass}>
      {ass.gradeUrl ? 
        <a href={ass.gradeUrl} target="_blank" rel="noopener noreferrer" >{ass.assignmentName}</a> : 
        <Link to="/ass"   >{ass.assignmentName}</Link>}
    </li>
  );
}

Ass.propTypes = {
  ass: PropTypes.object
};

export default Ass;
