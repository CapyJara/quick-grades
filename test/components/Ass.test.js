import React from 'react';
import { shallow } from 'enzyme';
import Ass from '../../src/components/asses/Ass';

describe('App component', () => {
  it('renders App', () => {
    const ass = {
      gradeUrl: 'http',
      assignmentName: 'todo'
    };
    const wrapper = shallow(<Ass ass={ass} />);
    expect(wrapper).toMatchSnapshot();
  });
});
