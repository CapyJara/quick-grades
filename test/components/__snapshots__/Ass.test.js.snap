// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<li
  className="ass"
>
  <a
    href="http"
    rel="noopener noreferrer"
    target="_blank"
  >
    todo
  </a>
</li>
`;
